lib/src/widgets/query.dart

```diff
@@ -62,22 +44,39 @@ class QueryState extends State<Query> {
       fetchResults: true,
       context: widget.options.context,
     );
+  }
+
+  void query(WatchQueryOptions options) {
```

Comment:
Maybe name the function `_initQuery`, and include the `shouldCreateNewObservable` logic from the old `didChangeDependencies` method.

Currently changes to the client will trigger `didChangeDependencies` which calls `query` directly and does not close any open streams in the `observableQuery`.

Reply:
good catch - added `observableQuery?.close()` to `_initQuery`. I left the `shouldCreateNewObservable` logic in `didUpdateWidget` because I figure if, for instance, the entire client changes, we should probably rebuild the query